cloud-agent/service/server/src/main/scala/org/hyperledger/identus/agent/server/jobs/StatusListJobs.scala

```diff
@@ -1,127 +1,179 @@
+          .mapError(_.underlying)
+        issuer <- createJwtIssuer(statusListWithCreds.issuer, VerificationRelationship.AssertionMethod)
+        vcStatusListCred <- VCStatusList2021
+          .decodeFromJson(vcStatusListCredJson, issuer)
+          .mapError(x => new Throwable(x.msg))
+        bitString <- vcStatusListCred.getBitString.mapError(x => new Throwable(x.msg))
+        _ <- ZIO.collectAll(
+          statusListWithCreds.credentials.map(c =>
+            updateBitStringForCredentialAndNotify(bitString, c, walletAccessContext)
+          )
+        )
+        unprocessedEntityIds = statusListWithCreds.credentials.collect {
+          case x if !x.isProcessed && x.isCanceled => x.id
+        }
+        _ <- credentialStatusListService
+          .markAsProcessedMany(unprocessedEntityIds)
+          @@ Metric
+            .gauge("revocation_status_list_sync_mark_as_processed_many_ms_gauge")
+            .trackDurationWith(_.toMetricsSeconds)
 
-          unprocessedEntityIds = statusListWithCreds.credentials.collect {
-            case x if !x.isProcessed && x.isCanceled => x.id
-          }
-          _ <- credentialStatusListService
-            .markAsProcessedMany(unprocessedEntityIds)
-            @@ Metric
-              .gauge("revocation_status_list_sync_mark_as_processed_many_ms_gauge")
-              .trackDurationWith(_.toMetricsSeconds)
+        updatedVcStatusListCred <- vcStatusListCred.updateBitString(bitString).mapError {
+          case VCStatusList2021Error.EncodingError(msg: String) => new Throwable(msg)
+          case VCStatusList2021Error.DecodingError(msg: String) => new Throwable(msg)
+        }
+        vcStatusListCredJsonString <- updatedVcStatusListCred.toJsonWithEmbeddedProof.map(_.spaces2)
+        _ <- credentialStatusListService.updateStatusListCredential(
+          statusListWithCreds.id,
+          vcStatusListCredJsonString
+        )
+      } yield ()
+      _ <- effect
+        .catchAll(e =>
+          ZIO.logErrorCause(s"Error processing status list record: ${statusListWithCreds.id} ", Cause.fail(e))
+        )
+        .catchAllDefect(d =>
+          ZIO.logErrorCause(s"Defect processing status list record: ${statusListWithCreds.id}", Cause.fail(d))
+        )
+        .provideSomeLayer(ZLayer.succeed(walletAccessContext))
+    } yield ()
+  }
 
-          updatedVcStatusListCred <- vcStatusListCred.updateBitString(bitString).mapError {
-            case VCStatusList2021Error.EncodingError(msg: String) => new Throwable(msg)
-            case VCStatusList2021Error.DecodingError(msg: String) => new Throwable(msg)
-          }
-          vcStatusListCredJsonString <- updatedVcStatusListCred.toJsonWithEmbeddedProof
-            .map(_.spaces2)
-          _ <- credentialStatusListService
-            .updateStatusListCredential(statusListWithCreds.id, vcStatusListCredJsonString)
-        } yield ()
+  private def updateBitStringForCredentialAndNotify(
```

Comment:
I suggest going ahead and decoupling the update and notifying operations with another topic between these ops.
Another consumer will process the notification using the retry logic in that case.
Let's talk about this at the AoH. It shouldn't block this PR.

Reply:
Yep, we should identify those processes that can be refactored to asynchronous. Kafka should be used across the board. Let's see how we handle this, given the recent restructuring.